### PR TITLE
Iteration three

### LINKED ISSUE #12: Iteration 3: Hydration Class
- [x] Take in user id
    - [x] Display data based on user id
- [x] For a user (identified by their userID - this is the same for all methods requiring a specific user’s data), the average fluid ounces consumed per day for all time
- [x] For a user, how many fluid ounces they consumed for a specific day (identified by a date)
- [x] For a user, how many fluid ounces of water consumed each day over the course of a week (7 days) - return the amount for each day

### LINKED ISSUE #13: Iteration 3: Hydration Test
### Will verify that:

- [x] Hydration should be a class
    - [x] Hydration should take in user id as the parameter
- [x] Hydration should have a method that calculates total average water consumed
- [x] Hydration should have a method that calculates daily average water consumed
- [x] Hydration should have a method that calculated weekly average water consumed

### LINKED ISSUE #14: Iteration 3: Hydration DOM
- [x] A display to show how much water they have consumed today (these displays are often called “widgets” in the FE tech world)
- [x] A display to show much water they have consumed each day over the course of the latest week

### PR DESCRIPTION
**Pull request by zliibbe**
# This pull request closes the following issues:
closes #12 
closes #13 
closes #14

This pull adds functionality to fetch and then display hydration data. Fetches all user data from https://fitlit-api.herokuapp.com/api/v1/hydration. Displays this data under hydration section of the page.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
    - Changed: 
    - Bugs created:
- [ ] Bug fix (non-breaking change which fixes an issue)
    -Bug fixed:
- [ ] Refactor
    -Code refactored:
- [ ] Documentation update

## How Has This Been Tested?

Tests written and passed using Mocha/Chai for the hydration class. 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have fully tested my code
- [x] My code follows the style guidelines of this project.
- [ ] My code has no unused/commented-out code.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

## Future Steps:

Complete iteration 4 per Turing spec.

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -13,6 +13,29 @@ class User {
...
+
+  totalAvgWater(hydration) {
+    let totalOunces = hydration.hydrationData.reduce((acc, x) => {
+      return acc + x.numOunces
+    }, 0)
+    return Math.round(totalOunces / hydration.hydrationData.length)
+  }
+
+  dailyWater(hydration, date) {
+    let dayWater = hydration.hydrationData.find(x => {
```
Comment (BrendaSotel0) [inline src/User.js:25]: dayWater? Sounds weird, might want to change that.

File: src/User.js [hunk 2]
```diff
@@ -13,6 +13,29 @@ class User {
     const name = this.name.split(' ');
     return name[0];
   }
+
+  totalAvgWater(hydration) {
+    let totalOunces = hydration.hydrationData.reduce((acc, x) => {
```
Comment (BrendaSotel0) [inline src/User.js:18]: What is x? What does it represent?

File: src/apiCalls.js [hunk 3]
```diff
@@ -6,11 +6,11 @@ const fetchUserData = () => {
   .then(response => response.json())}
 
 const fetchSleepData = () => {
-  fetch("https://fitlit-api.herokuapp.com/api/v1/sleep")
+  return fetch("https://fitlit-api.herokuapp.com/api/v1/sleep")
```
Comment (BrendaSotel0) [inline src/apiCalls.js:9]: Good catch on the returns

File: src/scripts.js [hunk 4]
```diff
@@ -34,6 +35,11 @@ let userFriends = document.querySelector('#user-friends');
 let moreInfoBtn = document.querySelector('#more-info');
 let infoDropdownContent = document.querySelector('#info-dropdown');
 let stepGoalComparison = document.querySelector('#step-goal-comparison');
+let hydrationTitle = document.querySelector('#hydration');
```
Comment (BrendaSotel0) [inline src/scripts.js:38]: Make sure all query selectors are used, delete any that are not needed

File: src/scripts.js [hunk 5]
```diff
@@ -16,6 +16,7 @@ console.log('This is the JavaScript entry file - your code begins here.');
 import userData from './data/users';
 import User from './User';
 import UserRepository from './UserRepository';
+import Hydration from './Hydration'
 
 //=====================================================
 //imports?
```
Comment (Andrew-Musselman) [inline src/scripts.js:22]: We have our imports above the space we labeled for imports, not a huge issue, but something to look at when we refactor

File: src/scripts.js [hunk 6]
```diff
@@ -53,34 +59,40 @@ let stepGoalComparison = document.querySelector('#step-goal-comparison');
...
-      luisa = allUsers.users[0]
-      console.log(allUsers.users[0])
-      generateUserInfoCard(luisa)
-    })
-
-    console.log(allUsers)
 
 
+const loadUserInfo = (id) => {
+    // let randomIndex = getRandomNumber(userData)
```
Comment (BrendaSotel0) [inline src/scripts.js:72]: Delete unneeded commented out code

File: src/scripts.js [hunk 7]
```diff
@@ -53,34 +59,40 @@ let stepGoalComparison = document.querySelector('#step-goal-comparison');
 
 let allUsers;
 let luisa;
+let hydrationUser;
+let ourUser;
 
 const getRandomNumber = (array) => {
     return Math.floor(Math.random() * array.length);
 }
```
Comment (Andrew-Musselman) [inline src/scripts.js:68]: It would be nice to get this random number function to work, if we can's we should remember to remove it when we refactor

**Pull request merged**